libraries/abstractions/secure_sockets/test/iot_test_tcp.c

```diff
@@ -1554,8 +1555,22 @@ static void prvSOCKETS_NonBlocking_Test( Server_t xConn )
                  */
                 xNumBytesReceived += lNumBytes;
             }
+            else if( ( lNumBytes < 0 ) && ( lNumBytes != SOCKETS_EWOULDBLOCK ) )
+            {
+                /* Since we are in non blocking mode, this error is expected if the TCP stack is
+                 * still processing the packet. */
```

Comment:
```suggestion
                /* Since we are in non blocking mode, SOCKETS_EWOULDBLOCK error is fine because the TCP stack may 
                 * still be processing the packet. Any other error is not expected and we break if we receive any other error. */
```